### DISCUSSION TITLE
Themes?

### DISCUSSION THREAD

**Discussion by razvan-nicolae** [vladmandic/sdnext#180] [General]
Any idea what we can use? I tried a few default ones but it's really bad, and I can't find the original 1111 anymore.
For some reason custom user.css doesn't work anymore either.

**Comment by vladmandic** [comment]
I'll check on why user.css is not used, it should be.
Regarding themes, few more steps and there's a theme builder...

**Comment by razvan-nicolae** [comment]
atm all i see is black-orange theme message on load of webui but that seems to supercede anything in the folder. And in config.json i was able to edit the black-orange theme but without further customization none are usable.

**Comment by vladmandic** [comment]
if there is actual issue, please create issue.
not sure what "none are usable" means? do they work? there is 11 themes other than `black-orange`? how is theme `gradio/default` not usable when that is default gradio theme without any skinning?
and if `user.css` is present, it will be loaded.

**Comment by razvan-nicolae** [comment]
Easy, the buttons are all stacked up in the wrong places with gradio/default. Aspect ratio for the UI is busted etc.

And in my case user.css which I was using from @FutonGama no longer works and it’s a really simple one so something definitely changed with themes update.

Sure it’s loaded but it is overwritten since you end up seeing orange theme not your user.css



On 17 Apr 2023, at 04:46, Vladimir Mandic ***@***.***> wrote:

﻿

if there is actual issue, please create issue.
not sure what "none are usable" means? do they work? how is theme gradio/default not usable when that is default gradio theme without any skinning?
and if user.css is present, it will be loaded.

—
Reply to this email directly, view it on GitHub<https://github.com/vladmandic/automatic/discussions/180#discussioncomment-5631439>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AESH55UA2EWCVZVN5T355L3XBSOEDANCNFSM6AAAAAAXALIORM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

_________________________________________________________________________________________________________________________

Ce message et ses pieces jointes peuvent contenir des informations confidentielles ou privilegiees et ne doivent donc
pas etre diffuses, exploites ou copies sans autorisation. Si vous avez recu ce message par erreur, veuillez le signaler
a l'expediteur et le detruire ainsi que les pieces jointes. Les messages electroniques etant susceptibles d'alteration,
Orange decline toute responsabilite si ce message a ete altere, deforme ou falsifie. Merci.

This message and its attachments may contain confidential or privileged information that may be protected by law;
they should not be distributed, used or copied without authorisation.
If you have received this email in error, please notify the sender and delete this message and its attachments.
As emails may be altered, Orange is not liable for messages that have been modified, changed or falsified.
Thank you.

**Comment by vladmandic** [comment]
`user.css` is loaded ON TOP of whatever theme is used. so if `black-orange` theme is selected, `user.css` will get loaded, but any values it does not define will remain from underlying theme. that is how any theme engine should work.

and regarding `gradio/default` being broken, can you share a screenshot? i don't see anything broken.
![image](https://user-images.githubusercontent.com/57876960/232483616-dce87a32-dffc-4a39-bb6c-6bc998f66abc.png)

**Comment by razvan-nicolae** [comment]
Looks fine now, no clue what was going on that particular commit I had when I tested

**Comment by ehamawy** [comment]
Hi, I'm having the same issue with the gradio/default theme.
Buttons are stacked too high, there is random whitespace, and the base model selector spans the whole width of the page
I've attached a screenshot for comparison
![vlad 11 bug](https://github.com/vladmandic/automatic/assets/9953712/23951c9b-7dc9-487d-924d-ddc7deb55f2c)

**Comment by vladmandic** [comment]
Well, thats Gradio default. You want to optimize it, go ahead. Or use one of actual optimized themes.
I've written in details on this topic several times.

**Reply by Woisek** [reply]
Is there also a list of what the actual SD:Next themes are? 🤔

**Reply by vladmandic** [reply]
first 6 - black-teal, black-orange, light-teal, invoked, amethyst-nightfall, midnight-barbie

**Reply by brknsoul** [reply]
Perhaps rename them `SD.Next-theme-name`?
